Userland/Libraries/LibWeb/Infra/Strings.cpp

```diff
@@ -29,4 +31,38 @@ String strip_and_collapse_whitespace(StringView string)
     return builder.string_view().trim(Infra::ASCII_WHITESPACE);
 }
 
+// https://infra.spec.whatwg.org/#code-unit-prefix
+bool is_code_unit_prefix(String const& potential_prefix, String const& input)
+{
+    auto potential_prefix_utf16 = Utf16View(utf8_to_utf16(potential_prefix));
+    auto input_utf16 = Utf16View(utf8_to_utf16(input));
```

Comment:
This looks suspiciously like UAF with the views pointing to temporary vectors... I think we can just omit `Utf16View` here and do `Utf16View(foo).code_unit_at()` in the loop below